tests/components/watchyourlan/test_config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Test the WatchYourLAN config flow."""
+
+from unittest.mock import AsyncMock, patch
+
+from httpx import ConnectError
+
+from homeassistant import config_entries
+from homeassistant.components.watchyourlan.const import DOMAIN
+from homeassistant.const import CONF_SSL, CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form and create entry with valid data."""
+    # Initiate the flow
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    # Check that the form is shown with no errors initially
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    # Patch the API call to simulate valid response
+    with patch(
+        "homeassistant.components.watchyourlan.config_flow.WatchYourLANClient.get_all_hosts",
+        new_callable=AsyncMock,
+        return_value={"title": "WatchYourLAN", "url": "http://127.0.0.1:8840"},
+    ):
```

Comment:
Consider creating a single mock for both the normal code as the config flow code. Checkout the `mock_airgradient_client` in `airgradient`